src/ZoneServer/Skills/Handlers/Highlander/WagonWheel.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Melia.Shared.Data.Database;
+using Melia.Shared.L10N;
+using Melia.Shared.Tos.Const;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Skills.Handlers.Highlander
+{
+	/// <summary>
+	/// Handler for the Highlander skill Wagon Wheel
+	/// </summary>
+	[SkillHandler(SkillId.Highlander_WagonWheel)]
+	public class WagonWheel : IGroundSkillHandler
+	{
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity target)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.SetAttackState(true);
+
+			var splashParam = skill.GetSplashParameters(caster, originPos, farPos, length: 70, width: 100, angle: 90);
+			var splashArea = skill.GetSplashArea(SplashType.Fan, splashParam); // DB states that this should be Square
+
+			Send.ZC_SKILL_READY(caster, skill, originPos, farPos);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, farPos, null);
+
+			this.Attack(skill, caster, splashArea);
+		}
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ISplashArea splashArea)
+		{
+			var hitDelay = TimeSpan.FromMilliseconds(50);
+			var damageDelay = TimeSpan.FromMilliseconds(100);
+			var skillHitDelay = TimeSpan.Zero;
+
+			await Task.Delay(hitDelay);
+
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+			var skillHits = new List<SkillHitInfo>();
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				var skillHitResult = SCR_SkillHit(caster, target, skill);
+				target.TakeDamage(skillHitResult.Damage, caster);
+
+				var skillHit = new SkillHitInfo(caster, target, skill, skillHitResult, damageDelay, skillHitDelay);
+				skillHit.ForceId = ForceId.GetNew();
+
+				skillHit.KnockBackInfo = new KnockBackInfo(caster.Position, target.Position, skill);
+				skillHit.HitInfo.Type = skill.Data.KnockDownHitType;
+				target.Components.Get<MovementComponent>().Stop();  // Temporary workaround, monsters that get knocked back should not be able to move for a while
```

Comment:
I get the idea of wanting to create a hotfix for a problem, but is this really much of a workaround? Whether stopping their movement does anything at all depends entirely on the AI, and they might start walking again right away. Adding temporary code for specific skills might just end up polluting the skill handlers. There should really be proper handling it instead.

Additionally, this code will throw an exception if a target doesn't have a movement component for some reason.

Reply:
You're right, and this doesn't really even work.  This definitely does need to be fixed elsewhere in code (being hit with a knockdown attack should presumably make you unable to do anything at all until you get up), but it's not the place to do it here.  I've removed it.